drivers/dai/intel/ssp/ssp.c

```diff
@@ -1586,14 +1687,37 @@ static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_co
 
 	ssp->state[DAI_DIR_PLAYBACK] = DAI_STATE_PRE_RUNNING;
 	ssp->state[DAI_DIR_CAPTURE] = DAI_STATE_PRE_RUNNING;
+}
 
-	/* ssp blob is set by pcm_hw_params for ipc4 stream, so enable
-	 * mclk and bclk at this time.
-	 */
-	dai_ssp_mn_set_mclk_blob(dp, blob->i2s_driver_config.mclk_config.mdivc,
-				 blob->i2s_driver_config.mclk_config.mdivr);
-	ssp->clk_active |= SSP_CLK_MCLK_ES_REQ;
+static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_config *cfg,
+				   const void *spec_config)
+{
+	const struct dai_intel_ipc4_ssp_configuration_blob_ver_1_5 *blob15 = spec_config;
+	const struct dai_intel_ipc4_ssp_configuration_blob *blob = spec_config;
```

Comment:
is there any platform relations between blob versions? When we can expect 1.5 or old one?

Reply:
they are unfortunately overlapping in tgl.

Reply:
so if I understand correctly, we should support always legacy blob and ver1.5 if needed tgl upwards

Reply:
yes, that makes sense